### PR TITLE
ZCS-2730:Adding resolve request parameter

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
Adding one more parameter in RestoreContactRequest called 'resolve' to support all 4 options (replace, ignore, reset and modify) in import API

### CODE REVIEW

**Comment by shrikant-prasad** [comment]
Please update soap.txt also.

**Comment by sneha-patil-synacor** [comment]
soap.txt updated.

File: soap/src/java/com/zimbra/soap/mail/message/RestoreContactsRequest.java [hunk 1]
```diff
@@ -35,6 +39,38 @@
     @XmlAttribute(name = MailConstants.A_CONTACTS_BACKUP_FILE_NAME, required = true)
     private String contactsBackupFileName;
 
+    /**
+     * @zm-api-field-tag resolve
+     * @zm-api-field-description restore resolve action
```
Comment (fatkudu) [inline soap/src/java/com/zimbra/soap/mail/message/RestoreContactsRequest.java]: Please state which is the default here - that is useful for the HTML SOAP docs.
Also, could you document what the various options mean please. I'm guessing ignore means if a contact with the same ID (what identifies a contact?) already exists, skip this one. modify means merge?? replace is fairly clear, no idea what reset means.  Although updating the enum with the docs makes most sense from a source perspective, currently here would be the best place to ensure the information ends up in the SOAP docs.

Comment (sneha-patil-synacor) [reply soap/src/java/com/zimbra/soap/mail/message/RestoreContactsRequest.java]: Updated the description. Please review https://github.com/Zimbra/zm-mailbox/pull/364

**Pull request closed without merging**